trollpy/tests.py

```diff
@@ -0,0 +1,357 @@
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    response = testapp.get('/registration', params={
+                           'csrf_token': csrf}, status=302).follow()
+    assert 'amosboldor@gmail.com' in response.html.find_all('li')[2].text
+    assert 'Amos Boldor' in response.html.find_all('li')[1].text
+
+
+def test_login_view_when_logged_in(testapp, fill_the_db):
+    """Test that registration view when logged in redirects."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    response = testapp.get('/login', params={
+                           'csrf_token': csrf}, status=302).follow()
+    assert 'amosboldor@gmail.com' in response.html.find_all('li')[2].text
+    assert 'Amos Boldor' in response.html.find_all('li')[1].text
+
+
+def test_registration_show_form(testapp):
+    """Test that the registration route shows form."""
+    response = testapp.get('/login', status=200)
+    assert response.status_code == 200
+    assert response.html.find('form')
+
+
+def test_profile_when_not_logged_in(testapp, fill_the_db):
+    """Test that profile works when not logged in."""
+    assert testapp.get('/profile/amos', status=200)
+
+
+def test_profile_route(testapp, fill_the_db):
+    """Test that the profile route pops up with correct info."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'}).follow()
+    response = testapp.get('/profile/amos', status=200)
+    assert 'amosboldor@gmail.com' in response.html.find_all('li')[2].text
+    assert 'Amos Boldor' in response.html.find_all('li')[1].text
+
+
+def test_profile_post(testapp, fill_the_db):
+    """Test that the profile route pops up with correct info."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'}).follow()
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    response = testapp.post('/profile/amos',
+                            params={'password': 'jdf;a',
+                                    'email': 'password@pe.com',
+                                    'first_name': 'Jeremy',
+                                    'last_name': 'renner',
+                                    'csrf_token': csrf}, status=302).follow()
+    assert 'password@pe.com' in response.html.find_all('li')[2].text
+    assert 'Jeremy renner' in response.html.find_all('li')[1].text
+
+
+def test_smack_api_route(testapp, fill_the_db):
+    """Test smack_api to return kill_score json."""
+    response = testapp.get('/smack_api', status=200)
+    assert len(response.json) == 1
```

Comment:
Again, what's inside the JSON that makes the response valid for your purposes, besides that there's the one item you expect?